addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -474,6 +474,33 @@ def test_mailing_shortener(self):
                     link_params=link_params,
                 )
 
+    @users('user_marketing')
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mailing_launch_flow(self):
+        """ Test that mail should be sent directly if total recipients are less than value
+        specified by config param 'mass_mailing_immediate_send_nbr', otherwise it should
+        enqueue the mailing """
+        mailing_1 = self.env['mailing.mailing'].create({
+            'name': 'One',
+            'subject': 'One',
+            'mailing_model_name': 'mailing.list',
+            'contact_list_ids': [(4, self.mailing_list_1.id), (4, self.mailing_list_2.id)],
+            'body_html': 'This is mass mail marketing demo'
+        })
+        mailing_1.action_launch()
+        self.assertEqual(mailing_1.state, 'done', "Mailing should be sent directly")
```

Comment:
You may use mock tools to mock emails (just to be sure), and I would also explicitely set the mass_mailing_immediate_send_nbr at the beginning of the method to make test explicit and independent from current value of config parameter. I see 3 scenarios
  * we are above the limit: queued
  * we are below limit: sent
  * limit is Falsy: queued

That way it is more explicit.